src/realmModels/Observation.js

```diff
@@ -408,13 +390,30 @@ class Observation extends Realm.Object {
       fields: { id: true }
     };
 
-    let response;
-    try {
-      response = await inatjs.observations.create( uploadParams, options );
-    } catch ( uploadError ) {
-      const errorText = await uploadError.response.text( );
-      uploadError.message = errorText;
-      throw uploadError;
+    const response = await createObservation( uploadParams, options );
+    await Observation.markRecordUploaded( obs.uuid, "Observation", response, realm );
+    const { id } = response.results[0];
+    if ( obs?.observationPhotos?.length > 0 ) {
+      return Observation.uploadEvidence(
+        obs.observationPhotos,
+        "ObservationPhoto",
+        ObservationPhoto.mapPhotoForUpload,
+        id,
+        inatjs.observation_photos,
+        realm,
+        options
+      );
+    }
+    if ( obs?.observationSounds?.length > 0 ) {
+      return Observation.uploadEvidence(
+        obs.observationSounds,
+        "ObservationSound",
+        ObservationSound.mapSoundForUpload,
+        id,
+        inatjs.observation_sounds,
+        realm,
+        options
+      );
```

Comment:
Again, no need to block, but if we're trying to get network stuff out of the model, maybe this entire method belongs in `useUploadObservations`.